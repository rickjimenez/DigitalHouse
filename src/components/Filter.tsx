import React, {FC, useEffect, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import Button from './Button';
import {FilterOptions} from '../types/products';

interface Props {
  onFilterChange?: (val: FilterOptions) => void;
}

/**
 * Filter Component
 *
 * @returns {React.ReactElement} Filter.
 */
const Filter: FC<Props> = ({onFilterChange = () => {}}): JSX.Element => {
  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.Todos);
  const styles = useStyles();

  useEffect(() => {
    onFilterChange(filter);
  }, [onFilterChange, filter]);

  return (
    <View style={styles.container}>
      {filter === FilterOptions.Todos ? (
        <>
          <Button
            label="Ganados"
            style={styles.leftBtn}
            onPress={() => setFilter(FilterOptions.Ganados)}
            testID="ganados-btn"
          />
          <Button
            label="Canjeados"
            style={styles.rightBtn}
            onPress={() => setFilter(FilterOptions.Canjeados)}
            testID="canjeados-btn"
          />
        </>
      ) : (
        <Button
          label="Todos"
          style={styles.rightBtn}
          onPress={() => setFilter(FilterOptions.Todos)}
          testID="todos-btn"
        />
      )}
    </View>
  );
};

const useStyles = () =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      marginTop: 20,
    },
    leftBtn: {
      marginRight: 10,
      flex: 1,
    },
    rightBtn: {
      flex: 1,
    },
  });

export default Filter;
