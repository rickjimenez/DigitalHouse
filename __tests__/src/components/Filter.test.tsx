import 'react-native';
import {render, screen, fireEvent} from '@testing-library/react-native';
import React from 'react';
import Filter from '../../../src/components/Filter';

describe('test Filter Comp', () => {
  test('renders correctly', () => {
    render(<Filter />);

    const comp = screen.toJSON();
    expect(comp).toMatchSnapshot();
  });

  test('test ganados btn', () => {
    const onFilterChange = jest.fn();
    render(<Filter onFilterChange={onFilterChange} />);
    const btn = screen.getByTestId('ganados-btn');
    fireEvent.press(btn);
    expect(onFilterChange).toBeCalled();
  });

  test('test canjeados btn', () => {
    const onFilterChange = jest.fn();
    render(<Filter onFilterChange={onFilterChange} />);
    const btn = screen.getByTestId('canjeados-btn');
    fireEvent.press(btn);
    expect(onFilterChange).toBeCalled();
    expect(screen.getByTestId('todos-btn')).toBeTruthy();
  });

  test('test todos btn', () => {
    const onFilterChange = jest.fn();
    render(<Filter onFilterChange={onFilterChange} />);
    const btn = screen.getByTestId('canjeados-btn');
    fireEvent.press(btn);
    expect(onFilterChange).toBeCalled();
    const todosBtn = screen.getByTestId('todos-btn');
    fireEvent.press(todosBtn);
    expect(screen.getByTestId('canjeados-btn')).toBeTruthy();
  });
});
