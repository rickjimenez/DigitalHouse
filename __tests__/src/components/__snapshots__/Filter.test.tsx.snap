// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Filter Comp renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 10,
        "opacity": 1,
      }
    }
    testID="ganados-btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "800",
        }
      }
    >
      Ganados
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#334FFA",
        "borderRadius": 10,
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="canjeados-btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "800",
        }
      }
    >
      Canjeados
    </Text>
  </View>
</View>
`;
